File: prismamate-backend/app/schemas/report.py
"""
PrismaMate 棱镜 - 报告 Schemas
"""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class ReportCreate(BaseModel):
    """报告创建 Schema"""
    task_id: int


class ReportResponse(BaseModel):
    """报告响应 Schema"""
    id: int
    report_id: str
    task_id: int
    report_hash: str
    pdf_url: Optional[str] = None
    verification_code: str
    created_at: datetime
    blockchain_tx_id: Optional[str] = None
    blockchain_status: Optional[str] = None

    class Config:
        from_attributes = True


class ReportVerifyResponse(BaseModel):
    """报告验证响应 Schema"""
    is_valid: bool
    report_id: str
    brand_names: list
    keywords: list
    platforms: list
    detection_time: str
    report_hash: str
    message: str

File: prismamate-frontend/src/api/report.ts
import api from './axios'

export interface Report {
  id: number
  report_id: string
  task_id: number
  report_hash: string
  pdf_url?: string
  verification_code: string
  created_at: string
  blockchain_tx_id?: string
  blockchain_status?: string
}

export interface VerifyResult {
  is_valid: boolean
  report_id: string
  brand_names: string[]
  keywords: string[]
  platforms: string[]
  detection_time: string
  report_hash: string
  message: string
}

export function createReport(taskId: number) {
  return api.post('/reports', { task_id: taskId })
}

export function getReport(reportId: string) {
  return api.get(`/reports/${reportId}`)
}

export function verifyReport(code: string) {
  return api.get(`/reports/verify/${code}`)
}

export function listReports(skip = 0, limit = 20) {
  return api.get('/reports', { params: { skip, limit } })
}

// 获取报告列表（兼容别名）
export const getReports = (skip = 0, limit = 50) => listReports(skip, limit)
